### ISSUE TITLE
`WindowMode::maximized` has no effect, unless `WindowMode::resizable` is set to `true`.

### ISSUE THREAD

**Issue by hannobraun** [ggez/ggez#732]
**Describe the bug**

`WindowMode::maximized` has no effect, unless `WindowMode::resizable` is set to `true`.

Note: I actually want `resizable` to be `true`, so this bug has no negative effect on me. I just noticed it, and thought I'd report it, hoping it will save someone else some trouble.

**To Reproduce**

Run this program (depends on ggez 0.5.1).
``` rust
use std::env;

use ggez::{
    Context,
    ContextBuilder,
    GameResult,
    conf::{
        FullscreenType,
        WindowMode,
    },
    event::{
        EventHandler,
        run,
    },
};


fn main() -> GameResult {
    // Workaround for https://github.com/ggez/ggez/issues/579.
    // I don't know, if this is required to reproduce this bug. It's
    // just the only way I found to get ggez working on my machine.
    env::set_var("WINIT_UNIX_BACKEND", "x11");

    let (mut context, mut event_loop) =
        ContextBuilder::new("window-maximize-bug", "Hanno Braun")
            .window_mode(
                WindowMode::default()
                    .fullscreen_type(FullscreenType::Windowed)
                    // This has no effect.
                    .maximized(true),
            )
            .build()?;

    let mut event_handler = Handler;

    run(&mut context, &mut event_loop, &mut event_handler)?;
    Ok(())
}


struct Handler;
```

**Expected behavior**

The program starts with a maximized window.

**Hardware and Software:**
 - ggez version: 0.5.1 (latest release, at the time of writing)
 - OS: Arch Linux
 - Graphics card: Intel UHD Graphics 620
 - Graphics card drivers: Whatever the default is. `lshw` says `driver=i915`

**Comment by icefoxen** [comment]
Confirmed, on Debian Linux using X11.  Thank you for the bug report!

**Comment by margorski** [comment]
Hello, 
I'm new into rust community and right now I'm looking for some opensource project to participate to get some experience. Your game framework is really nice and I thought about helping with easy issue and this one looks ideal.

I am confused about that one. Because Maximized attribute could be referring to different behaviour. 

In current code (src\graphics\context.rs:511):
` pub(crate) fn set_window_mode(&mut self, mode: WindowMode) -> GameResult {
        let window = &self.window;

        window.set_maximized(mode.maximized);`

`maximized` parameter is passed to `glutin::windowed::WindowedContext` method `set_maximized`. That makes sense that not resizable window cannot be maximized. 

But it could be fixed quickly by set resizable to true just before maximizing window and then reverting back option to config value.

`FullscreenType::Windowed => {
                window.set_fullscreen(None);
                window.set_decorations(!mode.borderless);
                window.set_inner_size(dpi::LogicalSize {
                    width: f64::from(mode.width),
                    height: f64::from(mode.height),
                });
                window.set_resizable(true);
                window.set_maximized(mode.maximized);
                window.set_resizable(mode.resizable);
            }`

But problem with that solution is that it cause that inner context is stretched (inner size is not adjusted) and after double clicking on window bar it will be resized back even if resizable is set to false.

If we want to have windowed fullscreen by setting maximized to true, then instead of setting `window.set_maximized` I believe that method window.set_inner_size should be used with dimensions calculated for current display. But that will change meaning of maximized parameter in relation to glutin windows.

I am not sure if that is understandable, but I just wonder if this is really bug. And if really feature for windowed fullscreen is needed then maybe new config parameter should be introduced to do not break consistency with glutin (and whole windows behaviour).

**Comment by PSteinhaus** [comment]
Just checked this. It seems to be fixed in 0.6.0-rc1.
Of course if someone is still able to reproduce this, please speak up and I'll reopen this issue.
